### PR TITLE
Hide Spatialite system tables

### PR DESCRIPTION
**Pull request by russss**
They were getting on my nerves.

### CODE REVIEW

**Comment by simonw** [comment]
We should only do this if we're certain the spatialite module has been loaded. I could imagine someone having a `sql_statements_log` table of their own without using spatialite for example.

I think the most reliable way to detect spatialite is to run `SELECT AddGeometryColumn(1, 2, 3, 4, 5);` against a `:memory:` database and see if it throws an exception - similar to how we detect FTS. We could add this as a `detect_spatialite()` function in `utils.py` and call it once on startup.

File: datasette/app.py [hunk 1]
```diff
@@ -1121,7 +1121,7 @@ def inspect(self):
                         tables[table]['foreign_keys'] = info
 
                     # Mark tables 'hidden' if they relate to FTS virtual tables
```
Comment (simonw) [inline datasette/app.py]: Need to update this comment

**Comment by russss** [comment]
> I think the most reliable way to detect spatialite is to run `SELECT AddGeometryColumn(1, 2, 3, 4, 5);` against a `:memory:` database and see if it throws an exception

Or just see if there's a `geometry_columns` table? I think that's quite unlikely to be added by accident (and it's an OGC standard). It also tells you if Spatialite is installed in the database rather than just loaded.

**Comment by simonw** [comment]
Nice, thanks very much.

**Pull request merged**